x-pack/test/functional/page_objects/lens_page.ts

```diff
@@ -197,7 +197,15 @@ export function LensPageProvider({ getService, getPageObjects }: FtrProviderCont
     },
 
     async searchField(name: string) {
-      await testSubjects.setValue('lnsIndexPatternFieldSearch', name);
+      const fieldSearchTestSubj = 'lnsIndexPatternFieldSearch';
+      await retry.try(async () => {
+        await testSubjects.setValue(fieldSearchTestSubj, '');
+        const val = await (await testSubjects.find(fieldSearchTestSubj)).getAttribute('value');
+        if (val !== '') {
+          throw new Error('not cleared yet');
+        }
+      });
+      await testSubjects.setValue(fieldSearchTestSubj, name);
```

Comment:
So I looked into the code of `testSubjects.setValue` and it's already doing a lot of logic to make sure it's actually cleared. For example, there is an option to clear with keyboard which you aren't using here. According to `test/functional/services/query_bar.ts`, clearWithKeyboard is more reliable than `clearValue`. I would suggest that we try that as the first step.